operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -62,63 +64,61 @@ public ConfigurationService getConfigurationService() {
    */
   @SuppressWarnings("unchecked")
   public void start() {
-    synchronized (lock) {
```

Comment:
don't we need to keep this lock ? if sop is invoked on a separate thread it may not perform any cleanup but controller may be running

Reply:
Do you have examples / use cases where `close` might be invoked asynchronously?

Reply:
🤔 Actually, thinking about it some more, if a shutdown hook is installed, then, indeed, `close` will be called from a different thread so we do need to re-instate synchronization.

Reply:
But that would be still just 1 thread. Is there still a situation where can be a race condition? By definition this should not be called by multiple threads.

Reply:
it depends who invokes the shutdown process, i.e. in spring & quarkus this is done by the underlying runtime ad we don't know if this happen in the same thread or not.

Reply:
If the operator app is stopped during the start process then it's possible that there would two different threads accessing the `started` and `controllers` fields at the same time.

Reply:
That is true.